### PR TITLE
Refactor Demo Style and System

### PR DESCRIPTION
**Pull request by kadiryazici**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
  - Refactor Demo System and Demo Components.

* **What is the current behavior?** (You can also link to an open issue here)
  - Current implementation is using TSX;
  - Pattern to create demos is poor.

* **What is the new behavior (if this is a feature change)?**
  - Current implementation is using Script Setup + TS + Vue;
  - Pattern to create demos is rich (i guess) now.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
  - Yes, now demo files should end with `.demo.ts`.

### CODE REVIEW

**Comment by kadiryazici** [comment]
There is a weird bug only on build, draft until I fix it.

File: vite.config.ts [hunk 1]
```diff
@@ -22,20 +25,31 @@ const productionConfig = defineConfig({
...
 });
 
 const demoConfig = defineConfig({
   root: path.join(root, 'demo'),
-  plugins: [VueJSX()],
+  plugins: [
+    VueJSX(),
+    Vue(),
+    WindiCSS({
+      config: path.join(root, 'tailwind.config.ts')
```
Comment (necatikcl) [inline vite.config.ts:44]: Windinin configi neden tailwind 🤔

Comment (kadiryazici) [reply vite.config.ts:44]: Windi hem tailwind hem windi.config.ts dosyalarını okuyabiliyor o yüzden ikisinin de ortası olsun diye yapmıştım.

Comment (kadiryazici) [reply vite.config.ts:44]: I should change it to windi :D thanks.

**Review by necatikcl — DISMISSED** [review]

**Review by necatikcl — APPROVED** [review]

**Pull request closed without merging**